### PR TITLE
[MODTLR-24] Create a new endpoint tlr/settings (GET, PUT methods) to manage ECS TLR settings

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Purpose
Create a new endpoint tlr/settings (GET, PUT methods) to manage ECS TLR settings. Response structure:

{
  "ecsTlrFeatureEnabled": boolean
}

Create an invisible (backend) permission tlr.settings.ecs-tlr.get for GET endpoint with display name “Settings (Circulation): Consortium title level requests view“. It has to include sub-permissions for the new endpoint described above as well as these:

settings.circulation.enabled
mod-settings.global.read.circulation
mod-settings.entries.collection.get
mod-settings.entries.item.get

Create a visible (visible: True) permission tlr.settings.ecs-tlr.put with display name “Settings (Circulation): Consortium title level requests edit“. It has to include sub-permissions for the new endpoint described above as well as these:

mod-settings.global.write.circulation
mod-settings.entries.item.put
mod-settings.entries.item.post

Resolves: [MODTLR-24](https://folio-org.atlassian.net/browse/MODTLR-24)

## Approach
- create endpoints;
- cover the new changes by unit tests;

### CODE REVIEW

File: src/main/java/org/folio/controller/EcsTlrController.java [hunk 1]
```diff
@@ -57,4 +58,24 @@ public ResponseEntity<Void> deleteEcsTlrById(UUID requestId) {
...
+    log.debug("getEcsTlrSettings:: ");
+
+    return ecsTlrService.getEcsTlrSettings()
+      .map(ResponseEntity.status(OK)::body)
+      .orElseGet(() -> ResponseEntity.notFound().build());
+  }
+
+  @Override
+  public ResponseEntity<Void> putEcsTlrSettings(EcsTlrSettings ecsTlrSettings) {
+    log.debug("putEcsTlrSettings:: parameters: {}", ecsTlrSettings);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/controller/EcsTlrController.java:73]:
```suggestion
    log.debug("putEcsTlrSettings:: parameters: {}", () -> ecsTlrSettings);
```

File: src/main/java/org/folio/service/impl/EcsTlrServiceImpl.java [hunk 2]
```diff
@@ -76,6 +83,30 @@ public boolean delete(UUID requestId) {
...
+  public Optional<EcsTlrSettings> getEcsTlrSettings() {
+    return ecsTlrSettingsRepository.findAll(PageRequest.of(0, 1))
+      .stream()
+      .findFirst()
+      .map(ecsTlrSettingsMapper::mapEntityToDto);
+  }
+
+  @Override
+  public Optional<EcsTlrSettingsEntity> updateEcsTlrSettings(EcsTlrSettings ecsTlrSettings) {
+    log.info("updateEcsTlrSettings:: parameters: {} ", () -> ecsTlrSettings);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/service/impl/EcsTlrServiceImpl.java:96]: debug?

File: src/test/java/org/folio/controller/EcsTlrControllerTest.java [hunk 3]
```diff
@@ -77,12 +79,68 @@ void ecsTlrShouldNotBeFound() {
...
+//  @Test
+//  void ecsTlrSettingsShouldSuccessfullyBeCreated() {
+//    var mockEcsTlrSettings = new EcsTlrSettings();
+//    when(ecsTlrService.createEcsTlrSettings(any(EcsTlrSettings.class))).thenReturn(mockEcsTlrSettings);
+//
+//    var response = ecsTlrController.postEcsTlrSettings(new EcsTlrSettings());
+//
+//    assertEquals(CREATED, response.getStatusCode());
+//    assertEquals(mockEcsTlrSettings, response.getBody());
+//  }
```
Comment (OleksandrVidinieiev) [inline src/test/java/org/folio/controller/EcsTlrControllerTest.java:127]: uncomment/remove?

File: src/main/java/org/folio/service/impl/EcsTlrSettingsServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    Optional<EcsTlrSettingsEntity> existentSettings = ecsTlrSettingsRepository.findAll(PageRequest.of(0, 1))
+      .stream()
+      .findFirst();
+
+    existentSettings.ifPresent(entity -> {
+      ecsTlrSettings.setId(entity.getId().toString());
+      ecsTlrSettingsRepository.save(ecsTlrSettingsMapper.mapDtoToEntity(ecsTlrSettings));
+    });
+
+    return existentSettings;
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/service/impl/EcsTlrSettingsServiceImpl.java:47]: Here we are returning the pre-update state of the settings. Can we return the post-update state?

```suggestion
    return ecsTlrSettingsRepository.findAll(PageRequest.of(0, 1))
      .stream()
      .findFirst()
      .map(entity -> ecsTlrSettingsRepository.save(
        ecsTlrSettingsMapper.mapDtoToEntity(ecsTlrSettings.id(entity.getId().toString()))));
```

File: src/main/java/org/folio/service/EcsTlrSettingsService.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package org.folio.service;
+
+import java.util.Optional;
+
+import org.folio.domain.dto.EcsTlrSettings;
+import org.folio.domain.entity.EcsTlrSettingsEntity;
+
+public interface EcsTlrSettingsService {
+  Optional<EcsTlrSettings> getEcsTlrSettings();
+  Optional<EcsTlrSettingsEntity> updateEcsTlrSettings(EcsTlrSettings ecsTlrSettings);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/service/EcsTlrSettingsService.java:10]: Can both of these methods return `Optional<EcsTlrSettingsEntity>`?

File: src/test/java/org/folio/service/EcsTlrServiceTest.java [hunk 6]
```diff
@@ -6,24 +6,28 @@
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
```
Comment (OleksandrVidinieiev) [inline src/test/java/org/folio/service/EcsTlrServiceTest.java:9]: A lot of new imports but no other changes?

**Review by alexanderkurash — APPROVED** [review]

**Review by OleksandrVidinieiev — APPROVED** [review]

**Pull request merged**